Source/WebCore/page/playstation/ResourceUsageThreadPlayStation.cpp

```diff
@@ -55,17 +56,33 @@ void ResourceUsageThread::platformCollectCPUData(JSC::VM*, ResourceUsageData& da
 
 void ResourceUsageThread::platformCollectMemoryData(JSC::VM* vm, ResourceUsageData& data)
 {
-    // FIXME: Calculate total dirty memory size for ResourceUsageOverlay
-    data.totalDirtySize = 0;
+    auto& categories = data.categories;
 
-    size_t currentGCHeapCapacity = vm->heap.blockBytesAllocated();
-    size_t currentGCOwnedExtra = vm->heap.extraMemorySize();
-    size_t currentGCOwnedExternal = vm->heap.externalMemorySize();
+    auto currentGCHeapCapacity = vm->heap.blockBytesAllocated();
+    auto currentGCOwnedExtra = vm->heap.extraMemorySize();
+    auto currentGCOwnedExternal = vm->heap.externalMemorySize();
     RELEASE_ASSERT(currentGCOwnedExternal <= currentGCOwnedExtra);
 
-    data.categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
-    data.categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
-    data.categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
+    categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
+    categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
+    categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
+
+    auto currentGCDirtySize = currentGCHeapCapacity + currentGCOwnedExtra - currentGCOwnedExternal;
+
+    // TODO: collect dirty size of "MemoryCategory::Images" and "MemoryCategory::Layers"
+
+    showmap::Result<256> result;
+    result.collect();
+    data.totalDirtySize = result.effectiveRss();
+
+    if (auto* entry = result.entry("SceNKFastMalloc")) {
+        auto rss = entry->effectiveRss();
+        if (data.totalDirtySize > rss)
+            categories[MemoryCategory::Other].dirtySize = data.totalDirtySize - rss;
+
+        categories[MemoryCategory::bmalloc] = rss - std::min(rss, currentGCDirtySize);
```

Comment:
This should come first than assigning to `Other`. Then `Other` = `totalDirtySize` - `bmalloc`.

Reply:
Then, the `currentGCDirtySize` would be included in `Other`. Is that correct?

Reply:
Sorry, but let me leave it as it is here.
If there is still a mistake, please point it out again.